### DISCUSSION TITLE
API Keys

### DISCUSSION THREAD

**Discussion by ghost** [ory/kratos#3236] [Q&A] [answered]
How to create & wotk with API keys, in self-hosted kratos??
I don't seem to find any way to get it, in documetation, 
There is only generated of api keys in ORY console 

https://www.ory.sh/docs/concepts/personal-access-token

**Comment by vinckr** [comment]
Hello @chintan-777 
That is correct, API Keys are just for Ory Network. 

Ory Kratos doesn't come with integrated access control. This means that all requests sent to the APIs are considered authenticated, authorized, and will be executed. 
When deploying Ory Kratos, protect access to their APIs using [Ory Oathkeeper](https://github.com/ory/oathkeeper) or a comparable API Gateway.

**Reply by ghost** [reply]
I mean, Admin authorization token (oryAccessToken)  to add identity
Can we get it, in self hosted environment? If yes, then how?

I want to use this end point - https://www.ory.sh/docs/reference/api#tag/identity/operation/createIdentity
with self-hosted kratos

**Reply by vinckr** [reply, marked as answer]
If you self-host Ory Kratos the admin endpoints do not have any authorization built in.

The oryAccessToken is a feature that is only present in Ory Network. 

You can replicate this feature when self-hosting by using [Ory Oathkeeper](https://github.com/ory/oathkeeper) or a comparable API Gateway.

**Reply by ghost** [reply]
okay 
Thanks a lot
